Add reference to prefer-ilm setting for apm lifeycle known bug (#4501)
elastic/observability-docs: docs/en/observability/apm/known-issues.asciidoc

lifecycle configurations.

// How to fix it
Upgrading to 8.15.1 resolves the lifecycle issue for any new indices created for APM data streams.
However, indices created in version 8.15.0 will remain unmanaged if the default ILM policy is in place.
To fix these unmanaged indices, consider one of the following approaches:

. Manually delete the unmanaged indices when they are no longer needed.
. Explicitly configure APM data streams to use the default data stream lifecycle configuration.
This approach migrates all affected data streams to use data stream lifecycles,
maintaining behavior equivalent to the default ILM policies.

> Apply this fix only to data streams that are currently unmanaged due to default ILM policies.

Author: ```suggestion
Apply this fix only to data streams that have unmanaged indices due to missing default ILM policies.
```

The datastream won't be unmanaged currently so a slight change of wordings.

Reviewer: Ah! Good catch.

. Manually delete the indices when they are no longer needed.
. Explicitly configure APM data streams with the default data stream lifecycle config.
Using this approach would migrate all data streams to use data stream lifecycles,
which should be equivalent to the default ILM policies. Note that if you have a custom
lifecycle policy in place for any of the datastreams then you don't need to do anything
for those datastreams as they are not affected by this bug. Going forward, for using

> custom ILM policies with new datastreams, please consult the {ref}/apm-ilm-how-to.html[updated guide].

Reviewer 2: Why do you think this belongs here? If I have a custom ILM policy I don't even read this line due to 
>The issue also does not occur if a custom ILM policy is configured using a custom component template.

Author: Took a stab at updating this in a different way to avoid confusion. PTAL.

Reviewer 2: This part looks good to me now - are you planning on also adding the `prefer-ilm` section to this PR or a separate one? (The PR description is not updated)
